extension/src/lib/domSettingsApplier.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * GPII Chrome Extension for Google Chrome
+ *
+ * Copyright 2017 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this license.
+ *
+ * You may obtain a copy of the license at
+ * https://github.com/GPII/gpii-chrome-extension/blob/master/LICENSE.txt
+ */
+
+/* eslint-env node */
+/* global fluid */
+
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+var chrome = chrome || fluid.require("sinon-chrome", require, "chrome");
+
+// This component makes use of css/Enactor.css to perform the adaptations
+// of the web content, and this is done through chrome.tabs.executeScript.
+//
+fluid.defaults("gpii.chrome.domSettingsApplier", {
+    gradeNames: ["fluid.modelComponent", "gpii.chrome.eventedComponent"],
+    domSettingsHandler: "content_scripts/domSettingsHandler.js",
+    events: {
+        onConnect: null
+    },
+    eventRelayMap: {
+        "chrome.runtime.onConnect": "onConnect"
+    },
+    model: {
+        // Accepted model values:
+        // settings.contrastTheme: string,
+        // settings.lineSpace: number,    // the multiplier to the current line space
+        // settings.inputsLarger: boolean,
+        // settings.selectionTheme: string,
+        // settings.simplifiedUiEnabled: boolean,
+        // settings.tableOfContentsEnabled: boolean
+    },
+    dynamicComponents: {
+        port: {
```

Comment:
Could there ever validly be more than one of these?

Reply:
Yes, a connection is created from every open tab and I believe also the prefsEditor.